requirements.txt

```diff
@@ -21,10 +21,7 @@ msvc-runtime==14.29.30133; sys.platform == 'win32' # append_to_freeze
 nibabel
 nilearn
 nnunetv2
-# This pin is due to an incompatibility with pystrum==0.2, which is installed via voxelmorph -> neurite -> pystrum
-# See also: https://github.com/adalca/pystrum/issues/9
-# If pystrum is updated to 0.3, then we should replace this version pinning with `pystrum>=0.3`
-numpy<1.24
```

Comment:
There's one last thing I'd like to do for this PR: Troubleshoot the new warnings that have arisen due to upgrading various packages (`numpy`, `pandas`, etc.)

On our Ubuntu 20.04 runner, this looks like:

```
============================== warnings summary ===============================
testing/api/test_deepseg_gm.py::TestCore::test_segment_volume
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/deepseg_/gm.py:231: RuntimeWarning: invalid value encountered in cast
    pred_slices = np.asarray(pred_slices, dtype=np.uint8)

testing/api/test_labels.py::test_continuous_vertebral_levels[test_image0]
testing/api/test_labels.py::test_continuous_vertebral_levels[test_image1]
testing/api/test_labels.py::test_continuous_vertebral_levels[test_image2]
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/labels.py:425: RuntimeWarning: invalid value encountered in scalar divide
    continuous_values[iz] = level + 2.0 * distance_from_level / float(length_levels[level])

testing/api/test_process_seg.py: 190 warnings
testing/cli/test_cli_sct_compute_compression.py: 567 warnings
testing/cli/test_cli_sct_process_segmentation.py: 170 warnings
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/process_seg.py:91: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
    angle_AP_rad = np.math.atan2(np.linalg.det([v0, v1]), np.dot(v0, v1))

testing/api/test_process_seg.py: 190 warnings
testing/cli/test_cli_sct_compute_compression.py: 567 warnings
testing/cli/test_cli_sct_process_segmentation.py: 170 warnings
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/process_seg.py:95: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
    angle_RL_rad = np.math.atan2(np.linalg.det([v0, v1]), np.dot(v0, v1))

testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements[dummy_lesion0-0.001]
testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements[dummy_lesion1-0.001]
testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements[dummy_lesion2-0.01]
testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements_without_segmentation[dummy_lesion0-0.001]
testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements_without_segmentation[dummy_lesion1-0.001]
testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements_without_segmentation[dummy_lesion2-0.01]
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_analyze_lesion.py:220: FutureWarning: save is not part of the public API, usage can give unexpected results and will be removed in a future version
    writer.save()

testing/cli/test_cli_sct_compute_compression.py: 36 warnings
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/pandas/core/algorithms.py:522: DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
  See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
    common = np.find_common_type([values.dtype, comps_array.dtype], [])

testing/cli/test_cli_sct_compute_compression.py::test_sct_compute_compression
testing/cli/test_cli_sct_compute_compression.py::test_sct_compute_compression
testing/cli/test_cli_sct_compute_compression.py::test_sct_compute_compression_sex_F
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/pandas/core/dtypes/cast.py:1641: DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
  See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
    return np.find_common_type(types, [])

testing/cli/test_cli_sct_label_vertebrae.py::test_sct_label_vertebrae_initial_disc_too_low
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/labels.py:133: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of -1 to uint8 will fail in the future.
  For the old behavior, usually:
      np.array(value).astype(dtype)
  will give the desired result (the cast overflows).
    out.data[x, y, z_rpi] = value

testing/cli/test_cli_sct_register_multimodal.py: 5 warnings
testing/cli/test_cli_sct_register_to_template.py: 152 warnings
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/numpy/core/_methods.py:198: RuntimeWarning: invalid value encountered in scalar divide
    ret = ret.dtype.type(ret / rcount)

testing/cli/test_cli_sct_warp_template.py::test_sct_warp_template_warp_full_PAM50
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/scipy/ndimage/_measurements.py:1541: RuntimeWarning: invalid value encountered in scalar divide
    results = [sum(input * grids[dir].astype(float), labels, index) / normalizer

testing/cli/test_sct_deepseg.py::test_segment_nifti[/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_testing_data/t2/t2.nii.gz-/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_testing_data/t2/t2_seg-manual.nii.gz-t2_seg_deepseg.nii.gz-seg_sc_contrast_agnostic-None]
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/torch/utils/data/dataloader.py:558: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
    warnings.warn(_create_warning_msg(

testing/cli/test_sct_deepseg.py::test_segment_nifti[/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_testing_data/t2/t2.nii.gz-/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_testing_data/t2/t2_seg-deepseg_rootlets.nii.gz-t2_seg_deepseg.nii.gz-seg_spinal_rootlets_t2w-None]
testing/cli/test_sct_deepseg.py::test_segment_nifti_multiclass[/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_testing_data/t2/t2_fake_lesion.nii.gz-fnames_seg_manual0-t2_deepseg.nii.gz-suffixes0-seg_sc_lesion_t2w_sci-0.5]
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/nnunetv2/utilities/plans_handling/plans_handler.py:37: UserWarning: Detected old nnU-Net plans format. Attempting to reconstruct network architecture parameters. If this fails, rerun nnUNetv2_plan_experiment for your dataset. If you use a custom architecture, please downgrade nnU-Net to the version you implemented this or update your implementation + plans.
    warnings.warn("Detected old nnU-Net plans format. Attempting to reconstruct network architecture "

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========== 440 passed, 13 skipped, 2065 warnings in 424.42s (0:07:04) ==========
```

I'll troubleshoot locally, then push some fixes where possible. :)

Reply:
Hrm. Several of these deprecation warnings stem from `numpy` -> `pandas` (i.e. `pandas` is out of date, but `numpy` is up to date).

These issues have been presumably fixed upstream by `pandas`, but something must be holding `pandas` back. Why? :thinking: 

Ah. We're on `pandas==1.5.3` right now, which is the last version prior to `2.0.0`. And, we're not on `pandas==2.0.0` [because of `ivadomed`](https://github.com/ivadomed/ivadomed/blob/1fccf77239985fc3be99161f9eb18c9470d65206/requirements.txt#L12-L13).

```python
# `pandas==2.0` removed the `DataFrame.append` call, which we currently use in approx. 13 places (per PyCharm search)
pandas>=1.1,<2.0
```

To fix this, we'll have to go through another cycle of "update ivadomed" -> "make a new ivadomed release" -> "update SCT", which is a bit cumbersome :sweat:. 

For now, I can just add some ignores to our `setup.cfg`?

Reply:
If you do add some temporary ignores to `setup.cfg`, can you also open an issue about it (and link it in the comments, ideally)?

Reply:
Done! :)